src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -31,14 +31,12 @@ internal class SslStreamInternal
         private int _nestedRead;
         private AsyncProtocolRequest _readProtocolRequest; // cached, reusable AsyncProtocolRequest used for read operations
         private AsyncProtocolRequest _writeProtocolRequest; // cached, reusable AsyncProtocolRequest used for write operations
+        private Action<Task, object> _freeBufferAction;
```

Comment:
Reading though, this is so the buffer can be freed with `.ContinueWith` so the buffer doesn't need to be tracked in class; and its a local rather than inline so it closure captures `this` once?

However `this` is only used for logging `this.GetHashCode()` in the free method, if logging is enabled; otherwise they could all be turned to statics? Though its only once per stream so probably doesn't matter hugely.